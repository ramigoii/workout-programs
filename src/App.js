import React, { useState } from "react";
import "./styles/styles.css";
import { Navbar, Footer } from "./Components";
import { Home, Programs, About, Contact, ProgramDetails, programs, programDetails } from "./MainPage";
const App = () => {
    const [activeComponent, setActiveComponent] = useState("home");
    const [selectedProgram, setSelectedProgram] = useState(null);

    // Переход к деталям программы
    const handleProgramClick = (program) => {
        setSelectedProgram({
            ...program,
            details: programDetails[program.id] || []
        });
    };
    const handleProg = () =>{
      setActiveComponent("programs");
    }
    // Возврат к списку программ
    const handleBackToPrograms = () => {
        setSelectedProgram(null);
    };
    const toContacts = () => {
      setActiveComponent("contact");
    }


    const renderComponent = () => {
        if (selectedProgram) {
            return <ProgramDetails program={selectedProgram} onBack={handleBackToPrograms} />;
        }

        switch (activeComponent) {
            case "home":
                return <Home OnProg={handleProg} OnContact={toContacts}/>;
            case "programs":
                return <Programs onProgramClick={handleProgramClick} />;
            case "about":
                return <About />;
            case "contact":
                return <Contact />;
            default:
                return <Home />;
        }
    };

    return (
        <>
            <Navbar setActiveComponent={setActiveComponent} />
            <div className="content">
                {renderComponent()}
            </div>
            <Footer />
        </>
    );
};

export default App;
