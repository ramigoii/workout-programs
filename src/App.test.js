// App.test.js
import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App, { ThemeProvider, useTheme } from './App';

// Вспомогательная функция для обёртки
const renderWithProviders = (ui) => {
  return render(
    <BrowserRouter>
      <ThemeProvider>
        {ui}
      </ThemeProvider>
    </BrowserRouter>
  );
};

// Тест ThemeProvider
test('ThemeProvider provides theme context', () => {
  const TestComponent = () => {
    const { theme } = useTheme();
    return <div data-testid="theme">{theme}</div>;
  };

  render(
    <ThemeProvider>
      <TestComponent />
    </ThemeProvider>
  );

  expect(screen.getByTestId('theme')).toHaveTextContent('light');
});

// Проверка рендера App
test('App renders navbar and main content', () => {
  renderWithProviders(<App />);
  expect(screen.getByText(/Workout Programs/i)).toBeInTheDocument();
});

// Снапшот тест
test('matches snapshot', () => {
  const { container } = renderWithProviders(<App />);
  expect(container).toMatchSnapshot();
});
