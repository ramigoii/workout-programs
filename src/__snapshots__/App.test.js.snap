// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <span>
          Workout
        </span>
         Programs
      </a>
    </div>
    <ul
      class="nav-links"
    >
      <li>
        <a
          class="nav-link active"
          href="/"
        >
          Главная
        </a>
      </li>
      <li>
        <a
          class="nav-link "
          href="/programs"
        >
          Программы
        </a>
      </li>
      <li>
        <a
          class="nav-link "
          href="/about"
        >
          О нас
        </a>
      </li>
      <li>
        <a
          class="nav-link "
          href="/contact"
        >
          Контакты
        </a>
      </li>
    </ul>
    <button
      aria-label="Переключить тему"
      class="theme-toggle"
      data-theme="light"
    >
      <svg
        class="moon-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
        />
      </svg>
    </button>
  </nav>
  <div
    class="content"
  >
    <div
      class="home-container"
    >
      <section
        class="hero"
      >
        <div
          class="hero-content"
        >
          <h1>
            Твой путь к идеальному телу
          </h1>
          <p>
            Выбирай программу и начни тренировки прямо сейчас
          </p>
          <button
            class="cta-button"
          >
            Выбрать программу
          </button>
        </div>
      </section>
      <section
        class="benefits"
      >
        <div
          class="container"
        >
          <div
            class="section-title"
          >
            <h2>
              Почему выбирают нас?
            </h2>
            <p>
              Наши преимущества, которые помогут вам достичь желаемых результатов
            </p>
          </div>
          <div
            class="benefit-items"
          >
            <div
              class="benefit-item"
            >
              <div
                class="benefit-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                  />
                </svg>
              </div>
              <h3>
                Персонализация
              </h3>
              <p>
                Индивидуальный подход к каждому клиенту
              </p>
            </div>
            <div
              class="benefit-item"
            >
              <div
                class="benefit-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 640 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M104 96H56c-13.3 0-24 10.7-24 24v104H8c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h24v104c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm528 128h-24V120c0-13.3-10.7-24-24-24h-48c-13.3 0-24 10.7-24 24v272c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V288h24c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM456 32h-48c-13.3 0-24 10.7-24 24v168H256V56c0-13.3-10.7-24-24-24h-48c-13.3 0-24 10.7-24 24v400c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V288h128v168c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24z"
                  />
                </svg>
              </div>
              <h3>
                Поддержка тренеров
              </h3>
              <p>
                Профессиональные тренеры всегда на связи
              </p>
            </div>
            <div
              class="benefit-item"
            >
              <div
                class="benefit-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 384 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M216 23.86c0-23.8-30.65-32.77-44.15-13.04C48 191.85 224 200 224 288c0 35.63-29.11 64.46-64.85 63.99-35.17-.45-63.15-29.77-63.15-64.94v-85.51c0-21.7-26.47-32.23-41.43-16.5C27.8 213.16 0 261.33 0 320c0 105.87 86.13 192 192 192s192-86.13 192-192c0-170.29-168-193-168-296.14z"
                  />
                </svg>
              </div>
              <h3>
                Доступ 24/7
              </h3>
              <p>
                Тренируйтесь в любое удобное время
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        class="programs-container"
      >
        <div
          class="container"
        >
          <div
            class="section-title"
          >
            <h2>
              Популярные программы
            </h2>
            <p>
              Выберите программу, которая подходит именно вам
            </p>
          </div>
          <div
            class="programs-grid"
          >
            <div
              class="program-card"
            >
              <img
                alt="Программа для похудения"
                src="/placeholder.svg"
              />
              <div
                class="program-card-content"
              >
                <h3>
                  Программа для похудения
                </h3>
                <p>
                  Эффективные тренировки для сжигания жира и подтянутого тела
                </p>
                <button
                  class="cta-button"
                >
                  Подробнее
                </button>
              </div>
            </div>
            <div
              class="program-card"
            >
              <img
                alt="Набор мышечной массы"
                src="strength.jpg"
              />
              <div
                class="program-card-content"
              >
                <h3>
                  Набор мышечной массы
                </h3>
                <p>
                  Силовые тренировки для увеличения мышечной массы
                </p>
                <button
                  class="cta-button"
                >
                  Подробнее
                </button>
              </div>
            </div>
            <div
              class="program-card"
            >
              <img
                alt="Йога и гибкость"
                src="yoga.jpg"
              />
              <div
                class="program-card-content"
              >
                <h3>
                  Йога и гибкость
                </h3>
                <p>
                  Гибкость и гармония тела с йогой и упражнениями на растяжку
                </p>
                <button
                  class="cta-button"
                >
                  Подробнее
                </button>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="testimonials"
      >
        <div
          class="container"
        >
          <div
            class="section-title"
          >
            <h2>
              Отзывы наших клиентов
            </h2>
            <p>
              Что говорят о нас те, кто уже добился результатов
            </p>
          </div>
          <div
            class="values-grid"
          >
            <div
              class="value-item"
            >
              <div
                class="value-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 496 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm0 376c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32zm0-128c-53.02 0-96 42.98-96 96s42.98 96 96 96c-106.04 0-192-85.96-192-192S141.96 64 248 64c53.02 0 96 42.98 96 96s-42.98 96-96 96zm0-128c-17.67 0-32 14.33-32 32s14.33 32 32 32 32-14.33 32-32-14.33-32-32-32z"
                  />
                </svg>
              </div>
              <h3>
                Анна, 28 лет
              </h3>
              <p>
                "Лучший сервис для тренировок! За 3 месяца я достигла результатов, о которых раньше могла только мечтать. "
              </p>
            </div>
            <div
              class="value-item"
            >
              <div
                class="value-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 640 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M104 96H56c-13.3 0-24 10.7-24 24v104H8c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h24v104c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm528 128h-24V120c0-13.3-10.7-24-24-24h-48c-13.3 0-24 10.7-24 24v272c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V288h24c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM456 32h-48c-13.3 0-24 10.7-24 24v168H256V56c0-13.3-10.7-24-24-24h-48c-13.3 0-24 10.7-24 24v400c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V288h128v168c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24z"
                  />
                </svg>
              </div>
              <h3>
                Дмитрий, 35 лет
              </h3>
              <p>
                "Очень удобно и эффективно! Тренировки подобраны идеально под мои цели, а поддержка тренеров всегда на высоте."
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        class="pricing"
      >
        <div
          class="container"
        >
          <div
            class="section-title"
          >
            <h2>
              Выберите свой тариф
            </h2>
            <p>
              Мы предлагаем различные варианты подписки для любых потребностей
            </p>
          </div>
          <div
            class="pricing-options"
          >
            <div
              class="card pricing-card"
            >
              <h3>
                Базовый
              </h3>
              <p>
                Доступ к базовым тренировкам
              </p>
              <span
                class="price"
              >
                999 руб/мес
              </span>
              <ul
                class="pricing-features"
              >
                <li>
                  30+ базовых тренировок
                </li>
                <li>
                  Доступ к форуму
                </li>
                <li>
                  Базовые рекомендации
                </li>
              </ul>
              <button
                class="cta-button"
              >
                Выбрать
              </button>
            </div>
            <div
              class="card pricing-card"
            >
              <h3>
                Премиум
              </h3>
              <p>
                Все тренировки + поддержка тренера
              </p>
              <span
                class="price"
              >
                1999 руб/мес
              </span>
              <ul
                class="pricing-features"
              >
                <li>
                  100+ тренировок
                </li>
                <li>
                  Поддержка тренера
                </li>
                <li>
                  Персональный план питания
                </li>
                <li>
                  Еженедельные отчеты
                </li>
              </ul>
              <button
                class="cta-button"
              >
                Выбрать
              </button>
            </div>
            <div
              class="card pricing-card"
            >
              <h3>
                VIP
              </h3>
              <p>
                Индивидуальный подход
              </p>
              <span
                class="price"
              >
                2999 руб/мес
              </span>
              <ul
                class="pricing-features"
              >
                <li>
                  Все тренировки
                </li>
                <li>
                  Персональный тренер 24/7
                </li>
                <li>
                  Индивидуальный план питания
                </li>
                <li>
                  Ежедневная поддержка
                </li>
                <li>
                  Анализ прогресса
                </li>
              </ul>
              <button
                class="cta-button"
              >
                Выбрать
              </button>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
  <footer
    class="footer"
  >
    <div
      class="footer-content"
    >
      <div
        class="footer-column"
      >
        <h3>
          Workout Programs
        </h3>
        <p>
          Лучшие фитнес-программы для вашего здоровья и тела
        </p>
        <div
          class="social-icons"
          style="margin-top: 1rem;"
        >
          <a
            class="social-icon"
            href="https://www.instagram.com/ramigoi/"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              />
            </svg>
          </a>
          <a
            class="social-icon"
            href="#"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm121.8 169.9l-40.7 191.8c-3 13.6-11.1 16.9-22.4 10.5l-62-45.7-29.9 28.8c-3.3 3.3-6.1 6.1-12.5 6.1l4.4-63.1 114.9-103.8c5-4.4-1.1-6.9-7.7-2.5l-142 89.4-61.2-19.1c-13.3-4.2-13.6-13.3 2.8-19.7l239.1-92.2c11.1-4 20.8 2.7 17.2 19.5z"
              />
            </svg>
          </a>
          <a
            class="social-icon"
            href="#"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M380.9 97.1C339 55.1 283.2 32 223.9 32c-122.4 0-222 99.6-222 222 0 39.1 10.2 77.3 29.6 111L0 480l117.7-30.9c32.4 17.7 68.9 27 106.1 27h.1c122.3 0 224.1-99.6 224.1-222 0-59.3-25.2-115-67.1-157zm-157 341.6c-33.2 0-65.7-8.9-94-25.7l-6.7-4-69.8 18.3L72 359.2l-4.4-7c-18.5-29.4-28.2-63.3-28.2-98.2 0-101.7 82.8-184.5 184.6-184.5 49.3 0 95.6 19.2 130.4 54.1 34.8 34.9 56.2 81.2 56.1 130.5 0 101.8-84.9 184.6-186.6 184.6zm101.2-138.2c-5.5-2.8-32.8-16.2-37.9-18-5.1-1.9-8.8-2.8-12.5 2.8-3.7 5.6-14.3 18-17.6 21.8-3.2 3.7-6.5 4.2-12 1.4-32.6-16.3-54-29.1-75.5-66-5.7-9.8 5.7-9.1 16.3-30.3 1.8-3.7.9-6.9-.5-9.7-1.4-2.8-12.5-30.1-17.1-41.2-4.5-10.8-9.1-9.3-12.5-9.5-3.2-.2-6.9-.2-10.6-.2-3.7 0-9.7 1.4-14.8 6.9-5.1 5.6-19.4 19-19.4 46.3 0 27.3 19.9 53.7 22.6 57.4 2.8 3.7 39.1 59.7 94.8 83.8 35.2 15.2 49 16.5 66.6 13.9 10.7-1.6 32.8-13.4 37.4-26.4 4.6-13 4.6-24.1 3.2-26.4-1.3-2.5-5-3.9-10.5-6.6z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="footer-column"
      >
        <h3>
          Навигация
        </h3>
        <ul
          class="footer-links"
        >
          <li>
            <a
              href="/"
            >
              Главная
            </a>
          </li>
          <li>
            <a
              href="/programs"
            >
              Программы
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              О нас
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Контакты
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-column"
      >
        <h3>
          Контакты
        </h3>
        <ul
          class="footer-links"
        >
          <li>
            Москва, ул. Примерная, 12
          </li>
          <li>
            +7 (900) 123-45-67
          </li>
          <li>
            support@workout.com
          </li>
        </ul>
      </div>
    </div>
    <div
      class="footer-bottom"
    >
      <p>
        © 2025 Workout Programs. Все права защищены. Сделано с
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="margin: 0px 0.25rem;"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
          />
        </svg>
      </p>
    </div>
  </footer>
</div>
`;
